integration-tests/src/test/java/org/wso2/appserver/test/integration/TestSuiteListener.java

```diff
@@ -0,0 +1,181 @@
+                System.setProperty(TestConstants.APPSERVER_PORT, String.valueOf(TestConstants.TOMCAT_DEFAULT_PORT));
+                applicationServerPort = Integer.valueOf(System.getProperty(TestConstants.APPSERVER_PORT));
+            }
+
+
+            log.info("Starting the server...");
+            applicationServerProcess = startPlatformDependApplicationServer();
+
+            waitForServerStartup();
+
+            if (isSuccessServerStartup) {
+                log.info("Application server started successfully. Running test suite...");
+            }
+
+        } catch (IOException ex) {
+            terminateApplicationServer();
+            String message = "Could not start the server";
+            log.error(message, ex);
+            throw new RuntimeException(message, ex);
+        }
+
+
+    }
+
+    @Override
+    public void onFinish(ISuite iSuite) {
+
+        log.info("Starting post-integration tasks...");
+        log.info("Terminating the Application server");
+        terminateApplicationServer();
+        log.info("Finished the post-integration tasks...");
+    }
+
+    private Process startPlatformDependApplicationServer() throws IOException {
+        String os = System.getProperty("os.name");
+        log.info(os + " operating system was detected");
+        if (os.toLowerCase().contains("unix") || os.toLowerCase().contains("linux")) {
+            log.info("Starting server as a " + os + " process");
+            return applicationServerProcess = new ProcessBuilder()
+                    .directory(appserverHome)
+                    .command("./bin/catalina.sh", "run")
+                    .start();
+        } else if (os.toLowerCase().contains("windows")) {
+            log.info("Starting server as a " + os + " process");
+            return applicationServerProcess = new ProcessBuilder()
+                    .directory(appserverHome)
+                    .command("\\bin\\catalina.bat", "run")
+                    .start();
+        }
+        return null;
+    }
+
+    public void terminateApplicationServer() {
+        if (applicationServerProcess != null) {
+            applicationServerProcess.destroy();
+        }
+    }
+
+    private void waitForServerStartup() throws IOException {
+
+        log.info("Checking server availability... (Timeout: " + serverStartCheckTimeout + " seconds)");
+        int currentSeconds = 0;
+        boolean isTimeout = false;
+        while (!isServerListening("localhost", applicationServerPort)) {
+            if (currentSeconds >= 20) {
+                isTimeout = true;
+                break;
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException ignored) {
+            }
+            currentSeconds++;
+        }
+
+        if (!isTimeout) {
+            isSuccessServerStartup = true;
+            log.info("Server started in " + currentSeconds + " seconds");
```

Comment:
As I see, the currentSeconds does not really correspond to number of seconds as per the above logic. Please correct it.